### DISCUSSION TITLE
Rename repo?

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre-style-spec#58] [Ideas]
Why is this repo called `maplibre-gl-style-spec` and not `maplibre-style-spec` while the style specification is called the MapLibre Style Specification?

I am asking because I am forseeing that MapLibre GL Native will be renamed to MapLibre Native at some point in the future.

**Comment by HarelM** [comment]
I don't have preference either way honestly...
@birkskyum?

**Comment by birkskyum** [comment]
I named the repo to align with the npm package it contains: https://www.npmjs.com/package/@maplibre/maplibre-gl-style-spec

**Comment by birkskyum** [comment]
It might make sense to keep the "MapLibre GL" as everything referring to the render, and "MapLibre" as the org name, to avoid some ambiguity. 

A similar argument to remove "Native" in the future could be made, in case a web build of the native repo one day is able replace gl js.

**Comment by louwers** [comment]
Maybe one day, but we have concrete plans for a Metal implementation this year, and as I understand it the GL in "MapLibre GL" refers to WebGL / OpenGL. And Native will depend on this repo.

**Comment by birkskyum** [comment]
GL is also used more generically for a range of graphics libraries https://vis.gl/frameworks 

None of them will be changing their names as a consequence of i.e. WebGPU

**Comment by birkskyum** [comment]
I was quite keen to erase the GL at some point to make for a cleaner naming scheme, as can be seen on https://github.com/maplibre/maplibre-native-base, so I'm not opposed to the idea, simply exploring if there are any downsides to the move, and where it would be a breaking change. Consistency is key

**Comment by louwers** [comment]
Alright. Could someone transfer this into a discussion? 👼

**Comment by birkskyum** [comment]
@louwers , closing in on publishing the style docs makes this quite relevant again... Since the repo contains both Utils and Spec, I even thought that simply "MapLibre Style" would be the right name for this repo. I agree that the GL is starting to feel more ceremonial than practical at this point, as I can see that Native etc. largely omit it.

**Reply by wipfli** [reply]
Ah now I understand why you chose the name `maplibre-style`. I associate the word `maplibre-style` with a map style like for example the demo tiles. On the other hand, when I hear `style spec`, then I think of the actual content of this repository. Maybe this is just me but I find the current name a bit misleading... Would you mind changing the name to `maplibre-style-spec`?

**Reply by wipfli** [reply]
Ah, and sorry to be late to this party...

**Reply by birkskyum** [reply]
Sure, will do.

**Reply by birkskyum** [reply]
PR here https://github.com/maplibre/maplibre-style/pull/115

**Comment by birkskyum** [comment]
Let's take this step now, so we don't have to redo it in near future.

**Comment by HarelM** [comment]
Please keep the npm library name the same for now. Feel free to change the repo name.

**Comment by louwers** [comment]
<img width="451" alt="image" src="https://user-images.githubusercontent.com/649392/231446092-99b06893-a026-4808-9e3e-52b88a44efaf.png">
